mesongenbuildreq.py

```diff
@@ -0,0 +1,29 @@
+import subprocess 
+import json 
+import sys
+deps_json = json.loads(subprocess.run([sys.argv[0], "introspect", "--dependencies", "meson.build"], capture_output=True).stdout) 
+deps = dict(zip([x['name'] for x in deps_json],[x['version'] for x in deps_json])) 
+deps.pop('', None) 
```

Comment:
Note that dependencies can be optional (e.g. `required: false`) but I guess the goal is to build-time-only require things that are plausibly needed even if not necessarily used by a given build (collecting runtime dependencies can be done by tracing DT_NEEDED tags)?